File: Synapse/backend/app/seed.py
"""
Seed the database with sample job postings.
Usage: python -m app.seed
"""
import asyncio
import uuid
from app.database import engine, async_session
from app.models import Profile, JobPosting
from app.middleware.auth import hash_password


SAMPLE_JOBS = [
    {
        "title": "Senior React Developer",
        "description": "We are looking for a Senior React Developer to join our frontend team. You will build and maintain high-performance web applications using React, TypeScript, and modern frontend tools.",
        "requirements": ["React", "TypeScript", "JavaScript", "HTML", "CSS", "REST APIs", "Git"],
        "responsibilities": ["Build responsive web applications", "Code review", "Mentor junior developers"],
        "location": "San Francisco, CA",
        "is_remote": True,
        "salary_min": 120000,
        "salary_max": 180000,
        "job_type": "full_time",
        "category": "Software Engineering",
    },
    {
        "title": "Python Backend Engineer",
        "description": "Join our backend team to build scalable APIs and microservices using Python, FastAPI, and PostgreSQL. Experience with cloud platforms preferred.",
        "requirements": ["Python", "FastAPI", "PostgreSQL", "REST APIs", "Docker", "AWS"],
        "responsibilities": ["Design and implement APIs", "Database optimization", "Write tests"],
        "location": "New York, NY",
        "is_remote": True,
        "salary_min": 110000,
        "salary_max": 160000,
        "job_type": "full_time",
        "category": "Software Engineering",
    },
    {
        "title": "Machine Learning Engineer",
        "description": "Build and deploy ML models for NLP and recommendation systems. Work with large datasets and production ML pipelines.",
        "requirements": ["Python", "TensorFlow", "PyTorch", "Scikit-learn", "SQL", "Docker", "AWS"],
        "responsibilities": ["Train and evaluate models", "Deploy models to production", "Data pipeline development"],
        "location": "Seattle, WA",
        "is_remote": False,
        "salary_min": 130000,
        "salary_max": 200000,
        "job_type": "full_time",
        "category": "Data Science & AI",
    },
    {
        "title": "Data Analyst & Business Intelligence",
        "description": "Analyze complex business datasets, design interactive Tableau/PowerBI dashboards, and deliver actionable data analytics to executive leadership.",
        "requirements": ["SQL", "Python", "Tableau", "Power BI", "Data Analysis", "Statistics", "Excel"],
        "responsibilities": ["Create executive reports", "Analyze customer retention trends", "Present data-driven findings"],
        "location": "Chicago, IL",
        "is_remote": True,
        "salary_min": 85000,
        "salary_max": 125000,
        "job_type": "full_time",
        "category": "Data Analytics",
    },
    {
        "title": "Business Strategy Consultant (MBA)",
        "description": "Drive corporate strategy, market expansion, and business transformation initiatives. Required MBA degree with strong strategic planning and leadership experience.",
        "requirements": ["MBA", "Business Strategy", "Financial Modeling", "Stakeholder Management", "Market Research", "Project Management"],
        "responsibilities": ["Lead strategic consulting projects", "Develop corporate growth roadmaps", "Brief senior executives"],
        "location": "New York, NY",
        "is_remote": False,
        "salary_min": 135000,
        "salary_max": 190000,
        "job_type": "full_time",
        "category": "Business & MBA",
    },
    {
        "title": "Senior Business Analyst",
        "description": "Bridge technology and business operational needs. Gather requirements, map process workflows, and evaluate business metrics for digital transformation.",
        "requirements": ["Business Analysis", "Requirement Gathering", "Agile/Scrum", "JIRA", "SQL", "Communication"],
        "responsibilities": ["Translate business needs to dev specs", "Facilitate stakeholder workshops", "Oversee UAT testing"],
        "location": "Boston, MA",
        "is_remote": True,
        "salary_min": 95000,
        "salary_max": 140000,
        "job_type": "full_time",
        "category": "Business & MBA",
    },
    {
        "title": "DevOps Engineer",
        "description": "Manage and improve our CI/CD pipelines, cloud infrastructure, and monitoring systems.",
        "requirements": ["Docker", "Kubernetes", "AWS", "CI/CD", "Terraform", "Linux", "Python"],
        "responsibilities": ["Maintain infrastructure", "Automate deployments", "Monitor system health"],
        "location": "Denver, CO",
        "is_remote": True,
        "salary_min": 100000,
        "salary_max": 150000,
        "job_type": "full_time",
        "category": "Cloud & DevOps",
    },
    {
        "title": "Financial & Risk Analyst",
        "description": "Perform quantitative financial analysis, portfolio valuation, and risk assessment for corporate investment decisions.",
        "requirements": ["Financial Modeling", "Corporate Finance", "Excel VBA", "SQL", "Bloomberg Terminal", "Accounting"],
        "responsibilities": ["Prepare financial forecasts", "Evaluate investment opportunities", "Monitor portfolio risk"],
        "location": "New York, NY",
        "is_remote": False,
        "salary_min": 90000,
        "salary_max": 130000,
        "job_type": "full_time",
        "category": "Finance & Accounting",
    },
    {
        "title": "Growth Marketing Manager",
        "description": "Develop and execute multi-channel digital marketing campaigns, performance analytics, SEO/SEM strategies, and customer acquisition funnels.",
        "requirements": ["Digital Marketing", "SEO/SEM", "Google Analytics", "Growth Hacking", "Content Strategy", "A/B Testing"],
        "responsibilities": ["Manage acquisition budgets", "Optimize conversion funnels", "Track marketing ROI"],
        "location": "Austin, TX",
        "is_remote": True,
        "salary_min": 85000,
        "salary_max": 120000,
        "job_type": "full_time",
        "category": "Marketing & Sales",
    },
]


async def seed():
    async with async_session() as session:
        # 1. Employer Account
        result = await session.execute(select(Profile).where(Profile.email == "employer@synapse.demo"))
        employer = result.scalars().first()
        
        if not employer:
            employer = Profile(
                id=uuid.uuid4(),
                email="employer@synapse.demo",
                full_name="Demo Employer",
                role="employer",
                company_name="TechCorp Inc.",
                password_hash=hash_password("Demo1234!"),
            )
            session.add(employer)
            await session.flush()

            for job_data in SAMPLE_JOBS:
                job = JobPosting(
                    employer_id=employer.id,
                    status="active",
                    **job_data,
                )
                session.add(job)
            print(f"Seeded employer (employer@synapse.demo / Demo1234!) with {len(SAMPLE_JOBS)} jobs.")
        else:
            print("Employer (employer@synapse.demo) already exists in database.")

        # 2. Seeker Account
        result_seeker = await session.execute(select(Profile).where(Profile.email == "seeker@synapse.demo"))
        seeker = result_seeker.scalars().first()

        if not seeker:
            seeker = Profile(
                id=uuid.uuid4(),
                email="seeker@synapse.demo",
                full_name="Demo Candidate",
                role="seeker",
                password_hash=hash_password("Demo1234!"),
            )
            session.add(seeker)
            await session.flush()
            print("Seeded seeker (seeker@synapse.demo / Demo1234!).")

        # 3. Admin Account
        result_admin = await session.execute(select(Profile).where(Profile.email == "admin@synapse.demo"))
        admin = result_admin.scalars().first()

        if not admin:
            admin = Profile(
                id=uuid.uuid4(),
                email="admin@synapse.demo",
                full_name="System Administrator",
                role="admin",
                company_name="Synapse Governance",
                password_hash=hash_password("Demo1234!"),
            )
            session.add(admin)
            await session.flush()
            print("Seeded admin (admin@synapse.demo / Demo1234!).")

        await session.commit()


if __name__ == "__main__":
    from sqlalchemy import select
    asyncio.run(seed())

File: Synapse/src/lib/seed.ts
import { api } from '@/lib/api-client';
import type { JobPosting } from '@/types';

const SAMPLE_JOBS: Array<{
  title: string;
  description: string;
  requirements: string[];
  responsibilities: string[];
  location: string;
  is_remote: boolean;
  salary_min: number;
  salary_max: number;
  job_type: string;
}> = [
  {
    title: 'Senior Frontend Engineer',
    description: 'We are looking for a Senior Frontend Engineer to lead our React-based web application development. You will architect user interfaces, mentor junior developers, and drive best practices in performance and accessibility.',
    requirements: ['React', 'TypeScript', 'Tailwind CSS', 'REST API', 'Git', '5+ years frontend experience'],
    responsibilities: ['Lead frontend architecture decisions', 'Build reusable component libraries', 'Optimize application performance', 'Mentor junior engineers'],
    location: 'San Francisco, CA',
    is_remote: true,
    salary_min: 140000,
    salary_max: 190000,
    job_type: 'full_time',
  },
  {
    title: 'Machine Learning Engineer',
    description: 'Join our AI team to build and deploy production ML models. You will work on NLP, computer vision, and recommendation systems.',
    requirements: ['Python', 'PyTorch', 'TensorFlow', 'Machine Learning', 'NLP', 'Docker', 'AWS'],
    responsibilities: ['Train and fine-tune ML models', 'Deploy models to production', 'Build data pipelines'],
    location: 'Remote',
    is_remote: true,
    salary_min: 130000,
    salary_max: 180000,
    job_type: 'full_time',
  },
  {
    title: 'Backend API Developer (Python)',
    description: 'Design and build scalable REST APIs using FastAPI and PostgreSQL. You will work on our core platform services.',
    requirements: ['Python', 'FastAPI', 'PostgreSQL', 'SQLAlchemy', 'Docker', 'REST API design'],
    responsibilities: ['Design API endpoints', 'Write clean, tested code', 'Optimize database queries'],
    location: 'Austin, TX',
    is_remote: true,
    salary_min: 110000,
    salary_max: 150000,
    job_type: 'full_time',
  },
  {
    title: 'Full-Stack JavaScript Developer',
    description: 'Build end-to-end features across our React + Node.js stack. You will own features from database schema to UI.',
    requirements: ['JavaScript', 'Node.js', 'React', 'Express', 'MongoDB', 'GraphQL'],
    responsibilities: ['Develop full-stack features', 'Write unit and integration tests', 'Deploy to production'],
    location: 'New York, NY',
    is_remote: false,
    salary_min: 100000,
    salary_max: 140000,
    job_type: 'full_time',
  },
  {
    title: 'DevOps & Cloud Engineer',
    description: 'Manage our cloud infrastructure on AWS, CI/CD pipelines, and container orchestration.',
    requirements: ['AWS', 'Docker', 'Kubernetes', 'Terraform', 'Jenkins', 'Linux', 'CI/CD'],
    responsibilities: ['Manage cloud infrastructure', 'Build CI/CD pipelines', 'Monitor system health'],
    location: 'Seattle, WA',
    is_remote: true,
    salary_min: 125000,
    salary_max: 170000,
    job_type: 'full_time',
  },
  {
    title: 'Data Scientist — Analytics',
    description: 'Analyze large datasets to drive business insights. You will build dashboards, run A/B tests, and create predictive models.',
    requirements: ['Python', 'SQL', 'Pandas', 'Tableau', 'Statistics', 'A/B Testing'],
    responsibilities: ['Analyze datasets for business insights', 'Build dashboards and reports', 'Run A/B experiments'],
    location: 'Boston, MA',
    is_remote: true,
    salary_min: 95000,
    salary_max: 130000,
    job_type: 'full_time',
  },
  {
    title: 'Product Designer (UI/UX)',
    description: 'Design intuitive, beautiful interfaces for our SaaS products.',
    requirements: ['Figma', 'Prototyping', 'User Research', 'Design Systems', 'HTML/CSS'],
    responsibilities: ['Design user interfaces', 'Conduct user research', 'Create prototypes'],
    location: 'Remote',
    is_remote: true,
    salary_min: 90000,
    salary_max: 125000,
    job_type: 'full_time',
  },
  {
    title: 'Junior Software Developer (Internship)',
    description: 'Great opportunity for recent grads to gain real-world experience.',
    requirements: ['JavaScript or Python', 'Git', 'Basic SQL', 'Eagerness to learn'],
    responsibilities: ['Build features under mentorship', 'Write tests', 'Learn best practices'],
    location: 'Chicago, IL',
    is_remote: true,
    salary_min: 50000,
    salary_max: 70000,
    job_type: 'internship',
  },
];

export async function seedSampleJobs(): Promise<void> {
  try {
    const existing = await api.get<JobPosting[]>('/api/v1/jobs?status=active&limit=1');
    if (existing.length > 0) return;

    for (const job of SAMPLE_JOBS) {
      await api.post('/api/v1/jobs', job);
    }
  } catch {
    // seed failed silently
  }
}

export async function getSampleJobsForFeed(): Promise<JobPosting[]> {
  return api.get<JobPosting[]>('/api/v1/jobs?status=active');
}
